packages/common-all/src/types/configs/publishing/publishing.ts

```diff
@@ -89,6 +92,52 @@ export type GoogleAnalyticsConfig = {
   tracking?: string;
 };
 
+export const publishingSchema = schemaForType<DendronPublishingConfig>()(
+  z
+    .object({
+      enableFMTitle: z.boolean().optional().default(true),
+      // enableHierarchyDisplay?: boolean;
+      // hierarchyDisplayTitle?: string;
+      enableNoteTitleForLink: z.boolean().optional().default(true),
+      enablePrettyRefs: z.boolean().optional().default(true),
+      // enableBackLinks?: boolean;
+      enableKatex: z.boolean().optional().default(true),
+      //
+      // assetsPrefix?: string;
+      copyAssets: z.boolean().default(true),
+      //
+      // canonicalBaseUrl?: string;
+      // customHeaderPath?: string;
+      // ga?: GoogleAnalyticsConfig;
+      // logoPath?: string;
+      // siteFaviconPath?: string;
+      // siteIndex?: string;
+      siteHierarchies: z.array(z.string()).default(["root"]),
+      writeStubs: z.boolean().default(false),
+      siteRootDir: z.string().default("docs"),
+      seo: seoSchema,
+      github: githubSchema,
+      enableSiteLastModified: z.boolean().default(true),
+      // siteUrl?: string;
+      enableFrontmatterTags: z.boolean().default(true),
+      enableHashesForFMTags: z.boolean().default(false),
+      enableRandomlyColoredTags: z.boolean().optional().default(true),
+      enableTaskNotes: z.boolean().optional().default(true),
+      // hierarchy?: { [key: string]: HierarchyConfig };
+      // duplicateNoteBehavior?: DuplicateNoteBehavior;
+      // theme?: Theme;
+      // segmentKey?: string;
+      // cognitoUserPoolId?: string;
+      // cognitoClientId?: string;
+      enablePrettyLinks: z.boolean().default(true),
+      // siteBanner?: string;
+      // giscus?: GiscusConfig;
+      // sidebarPath?: string | false;
+      searchMode: searchModeSchema.optional().default(SearchMode.LOOKUP),
+    })
+    .passthrough()
+);
```

Comment:
Now we define the default values in 2 locations. One option would be to use the schema inside `genDefaultPublishingConfig` like so: 

```ts
export function genDefaultPublishingConfig(): DendronPublishingConfig {
  const defaultConfigResult = parse(publishingSchema, {})
  if (defaultConfigResult.isErr()) {
    throw new DendronError({
      message: ErrorMessages.formatShouldNeverOccurMsg(),
    });
  }
  return defaultConfigResult.value
}
```

This creats a small overhead since now we need to parse for accesing the default values. We could do this only once the file gets imported.

@kevinslin @jonathanyeung @hikchoi

Reply:
@hikchoi has plans of a config service. we can do the parsing once when that service is in place. right now - i would just definite it in two places and make sure `[[dendron://dendron.docs/pkg.common-all.dev.cook.add-new-config]]` is updated

Reply:
Maybe we should rename the "term" schema, as it clashes with our exsiting semantic meaning with "Dendron Schemas".

Reply:
can we refer to types enforced by zod as zod schemas vs dendron schemas?